### PR TITLE
Added GLM(i think?) and a free cam class

### PR DESCRIPTION
**Pull request by matty2048**
Added glm.
Created a free cam with controls (WASD and mouse) for movement and looking around.
 created a view matrix but not a projection one.

### CODE REVIEW

File: build/vs2019/FreeCam.h [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+			freecamGroup->CreateAxisInputBinding("lookRight", 0)->BindCallback(std::bind(&FreeCam::LookRight, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("forward", GLFW_KEY_W)->BindCallback(std::bind(&FreeCam::MoveForward, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("backward", GLFW_KEY_S)->BindCallback(std::bind(&FreeCam::MoveBackward, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("left", GLFW_KEY_A)->BindCallback(std::bind(&FreeCam::MoveLeft, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("right", GLFW_KEY_D)->BindCallback(std::bind(&FreeCam::MoveRight, this, std::placeholders::_1));
+		}
```
Comment (Gaztin) [inline build/vs2019/FreeCam.h:29]: Avoid implementations in header files, unless they are one-liners and single-instruction 😊

File: build/vs2019/FreeCam.h [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+		glm::vec3 UpDir = glm::vec3(0,1,0); //direction of up for the camera
+		glm::vec3 FrontDir = glm::vec3(0,0,1); //direction the camera is pointing
+		glm::vec3 CameraPos = glm::vec3(0,0,-3); //position of the camera in the world
+
+		float CameraSpeed = 1; //speed of the camera
+		float Sensitivity = 0.1f; //sensitivity of the look
+
+
+		void UpdateView();
+		glm::mat4 ViewMatrix; //the view matrix of the camera
```
Comment (Gaztin) [inline build/vs2019/FreeCam.h:60]: Kinda hard to read. Would appreciate it if the functions were bundled together, and the variables bundled together. Also, please follow the naming convention 👍

File: src/Game/Game.cpp [hunk 3]
```diff
@@ -40,6 +40,7 @@ namespace gp1 {
 		m_Logger.LogDebug("CloseMenu keybind is: %u, on device: %u", closeMenu->GetIndex(), (uint32_t)closeMenu->GetLocation());
 	}
 
+
```
Comment (Gaztin) [inline src/Game/Game.cpp:43]: How did this get here?

**Review by Gaztin — CHANGES_REQUESTED** [review]

**Comment by Gaztin** [comment]
[Camera Controller](https://trello.com/c/eZDAvXka/24-camera-controller)

File: src/Game/Game.cpp (hunk 3 above)
Comment (matty2048) [reply src/Game/Game.cpp:43]: i have no idea....

**Review by Gaztin — APPROVED** [review]

**Pull request merged**